app/node.js

```diff
@@ -72,34 +99,23 @@ async function main() {
       console.error(
         `${args.knownHost}:${args.knownPort} is not responsive. Exiting`
       );
+      console.error("here");
       process.exit(-9);
     } else {
       console.log(`${args.knownHost}:${args.knownPort} responded`);
     }
   }
-
-  // protect against bad ID inputs
-  if (args.id && args.id > 2 ** HASH_BIT_LENGTH - 1) {
-    console.error(
-      `Error. Bad ID {${args.id}} > 2^m-1 {${2 ** HASH_BIT_LENGTH -
-        1}}. Terminating...\n`
-    );
-    return -13;
-  }
-
-  // protect against bad Known ID inputs
-  if (args.knownId && args.knownId > 2 ** HASH_BIT_LENGTH - 1) {
-    console.error(
-      `Error. Bad known ID {${args.knownId}} > 2^m-1 {${2 ** HASH_BIT_LENGTH -
-        1}}. Thus, terminating...\n`
-    );
-    return -13;
-  }
+  */
 
   try {
     userServiceNode = new UserService({ ...args });
     await userServiceNode.serve();
-    await userServiceNode.joinCluster();
+    let knownNode = {
+      id: knownNodeId,
+      host: knownNodeHost,
+      port: knownNodePort
+    };
+    await userServiceNode.joinCluster(knownNode);
```

Comment:
This is nice! The API makes much more sense shifting these values to the joinCluster method. Out of curiosity, why do we need `knownNodeId` at this point? Wouldn't we ask this known node for this information in all circumstances?

Reply:
I just put a comment in with a lengthy explanation but then accidentally delete it.  Bottom line is, yes, I tend to agree with you even though it was an oversight that I forgot to put the logic of handling the argument-provided knownId.  I'll go ahead and merge "as is" but happy to either remove the argument or add the logic if any team member wants to keep that option.  I definitely think we want to keep the option of forcing the "self" ID but can't see why we wouldn't always just ask the known node, as we are doing currently in join().